File: dsl-style-java/src/test/java/example/dsl_style_java/BatchUpdateTest.java
package example.dsl_style_java;

import example.dsl_style_java.domain.Salary;
import example.dsl_style_java.repository.EmployeeRepository;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.seasar.doma.jdbc.Config;

@ExtendWith(TestEnvironment.class)
public class BatchUpdateTest {

  private final EmployeeRepository repository;

  BatchUpdateTest(Config config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testBatchUpdate() {
    var list = repository.selectAll();
    for (var employee : list) {
      var salary = employee.getSalary();
      if (salary != null) {
        employee.setSalary(salary.add(new Salary(100)));
      }
    }
    repository.batchUpdate(list);
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl/style/kotlin/BatchUpdateTest.kt
package example.dsl.style.kotlin

import example.dsl.style.kotlin.domain.Salary
import example.dsl.style.kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config

@ExtendWith(TestEnvironment::class)
class BatchUpdateTest(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testBatchUpdate() {
        val list = repository.selectAll()
        for (employee in list) {
            val salary = employee.salary
            if (salary != null) {
                employee.salary = salary.add(Salary(100))
            }
        }
        repository.batchUpdate(list)
    }
}
